applications/MetisApplication/tests/test_metis_submodelpart_list.py

```diff
@@ -41,23 +41,25 @@ def test_ReadTwoSubModelParts(self):
             },
             "echo_level" : 0
         }""")
-        results = {"Main.submodelpart_liquid" : [133, 381, 228],
-                   "Main.submodelpart_solid" :  [280, 810, 552]}
+        results = [["Main.submodelpart_liquid" , [133, 381, 228]],
+                   ["Main.submodelpart_solid" ,  [280, 810, 552]]]
         ReadModelPart(self.file_name, model_part, settings)
         for submodel_part_name in results:
+            submodel_part_name = i_result[0]

```

Comment:
i think you also need to change the for...

~~~
for i_result in results:
~~~